internal/operators/mce/mce_operator.go

```diff
@@ -29,13 +29,12 @@ var Operator = models.MonitoredOperator{
 }
 
 // NewMceOperator creates new MCE operator.
-func NewMceOperator(log logrus.FieldLogger) *operator {
-	cfg := Config{}
-	err := envconfig.Process(common.EnvConfigPrefix, &cfg)
+func NewMceOperator(log logrus.FieldLogger, config Config) *operator {
+	err := envconfig.Process(common.EnvConfigPrefix, &config)
 	if err != nil {
 		log.Fatal(err.Error())
 	}
-	return newMceOperatorWithConfig(log, &cfg)
+	return newMceOperatorWithConfig(log, &config)
```

Comment:
I think we should unmarshall the config here, so we can fail the startup of the service quickly if the config is wrong. Also, it's worth to unmarshall it only once, not each time we call `getMCEVersion()`

Reply:
I guess `NewMceOperator` is called when the assisted-service starts, right?

Reply:
> I guess NewMceOperator is called when the assisted-service starts, right?

yes, through `NewManager` https://github.com/danmanor/assisted-service/blob/master/cmd/main.go#L339C1-L339C119

Reply:
Ack.